libfoundation/src/foundation-locale.cpp

```diff
@@ -1143,10 +1143,15 @@ bool MCLocaleWordBreakIteratorAdvance(MCStringRef self, MCBreakIteratorRef p_ite
         // if the intervening chars contain a letter or number then it was a valid 'word'
         while (t_left_break < t_right_break)
         {
-            if (MCStringCodepointIsWordPart(MCStringGetCodepointAtIndex(self, t_left_break)))
-                break;
-            if (MCStringIsValidSurrogatePair(self, t_left_break++))
-                t_left_break++;
+			codepoint_t t_cp =
+				MCStringGetCodepointAtIndex(self, t_left_break);
+			
+			if (MCStringCodepointIsWordPart(t_cp))
+				break;
+			
+			t_left_break++;
+			if (t_cp > UNICHAR_MAX)
+				t_left_break++;
```

Comment:
Hmm... actually, is this a sufficiently common pattern that there ought to be an inline helper function?

```C++
t_left_break += MCUnicodeCodepointNumChars(t_cp);
```

or something similar?